include/cetl/any.hpp

```diff
@@ -49,41 +53,97 @@ struct base_storage  // NOLINT(*-pro-type-member-init)
     template <typename Tp>
     void make_handlers() noexcept
     {
-        assert(nullptr == value_destroyer_);
+        static_assert(sizeof(Tp) <= Footprint, "Enlarge the footprint");
+
+        CETL_DEBUG_ASSERT(nullptr == value_destroyer_, "Expected to be empty before making handlers.");
+        CETL_DEBUG_ASSERT(nullptr == value_converter_, "");
+        CETL_DEBUG_ASSERT(nullptr == value_const_converter_, "");
+        CETL_DEBUG_ASSERT(nullptr == value_type_informer_, "");
 
         value_destroyer_ = [](void* const storage) {
             const auto ptr = static_cast<Tp*>(storage);
             ptr->~Tp();
         };
+
+        make_converters<Tp>();
+
+        value_type_informer_ = [](const void*) {
+
+#if __cpp_rtti
```

Comment:
Here, we are stepping on a slippery slope of relying on implementation-specific features. In general, as far as I know, the C++ standard provides no feature test macro for RTTI:

<img width="766" alt="image" src="https://github.com/OpenCyphal/CETL/assets/3298404/40ee883e-aa25-4972-8e89-e57b58f8e566">

I would limit our implementation of `any` to CETL RTTI only.

From our earlier conversations, I understood that you were adding the type informer handler `value_type_informer_` in order to support CETL RTTI rather than the standard `typeid`.

Reply:
ok, I didn't know that `__cpp_rtti` is some kind of an extension. I will remove related stuff and stick with our CETL RTTI only. Could you please give a link (or pdf) from where you made the screenshot - for future reference. Thnx!

Reply:
here: https://github.com/cplusplus/draft/releases/tag/n4917